includes/class-regeneratethumbnails-regenerator.php

```diff
@@ -180,6 +180,8 @@ public function get_fullsizepath() {
 	public function regenerate( $args = array() ) {
 		global $wpdb;
 
+		do_action( 'regenerate_thumbnails_regenerator', $this->attachment->ID );
```

Comment:
Can you rename it to: `regenerate_thumbnails_regenerator_pre_regenerate` ?

The idea is to be as descriptive as possible with when the hook is being triggered: before the main regenerate logic in the regenerator class.